core/src/main/java/org/apache/iceberg/ReachableFileUtil.java

```diff
@@ -148,12 +148,49 @@ public static List<String> statisticsFilesLocations(Table table) {
    * @param table table for which statistics files needs to be listed
    * @param predicate predicate for filtering the statistics files
    * @return the location of statistics files
+   * @deprecated use the {@code statisticsFilesLocationsForSnapshots(table, snapshotIds)} instead.
    */
+  @Deprecated
   public static List<String> statisticsFilesLocations(
       Table table, Predicate<StatisticsFile> predicate) {
     return table.statisticsFiles().stream()
         .filter(predicate)
         .map(StatisticsFile::path)
         .collect(Collectors.toList());
   }
+
+  /**
+   * Returns locations of all statistics files for a table matching the given snapshot IDs.
+   *
+   * @param table table for which statistics files needs to be listed
+   * @param snapshotIds ids of snapshots for which statistics files will be returned. If null,
+   *     statistics files for all the snapshots will be returned.
+   * @return the location of statistics files
+   */
+  public static List<String> statisticsFilesLocationsForSnapshots(
```

Comment:
What do you think about calling it `statisticsFilesLocations` to match `manifestListLocations` that also accepts a set of snapshot IDs?

Reply:
The deprecated class already has a name `statisticsFilesLocations`. So, If I keep the same name, compiler says ambiguous error when I call `statisticsFilesLocationsForSnapshots(table, null)`

Hence, I had to keep different name.